### PR TITLE
Format options for doctest directives

### PR DESCRIPTION
**Pull request by timhoffm**
Convert to standard syntax conventions we use for directive options.

### CODE REVIEW

File: doc/usage/extensions/doctest.rst [hunk 1]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
...
-     doctest flags (comments looking like ``# doctest: FLAG, ...``) at the
-     ends of lines and ``<BLANKLINE>`` markers are removed (or not removed)
-     individually.  Default is ``trim-doctest-flags``.
+         Supported PEP-440 operands and notations
 
-   Note that like with standard doctests, you have to use ``<BLANKLINE>`` to
-   signal a blank line in the expected output.  The ``<BLANKLINE>`` is removed
-   when building presentation output (HTML, LaTeX etc.).
+   .. rst:directive:option:: trim-doctest-flags
+   .. rst:directive:option:: no-trim-doctest-flags
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
   .. rst:directive:option:: trim-doctest-flags
                             no-trim-doctest-flags
```

File: doc/usage/extensions/doctest.rst [hunk 2]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
...
-   This directive supports three options:
+   .. rubric:: Options
+
+   .. rst:directive:option:: hide
+
+      Hide the code block in other builders.  By default it is shown as a
+      highlighted code block.
+
+   .. rst:directive:option:: trim-doctest-flags
+   .. rst:directive:option:: no-trim-doctest-flags
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
   .. rst:directive:option:: trim-doctest-flags
                             no-trim-doctest-flags
```

File: doc/usage/extensions/doctest.rst [hunk 3]
```diff
@@ -159,18 +200,28 @@ a comma-separated list of group names.
...
+      :type: comma separated list
 
-   * ``trim-doctest-flags`` and ``no-trim-doctest-flags``, a flag option,
-     doctest flags (comments looking like ``# doctest: FLAG, ...``) at the
-     ends of lines and ``<BLANKLINE>`` markers are removed (or not removed)
-     individually.  Default is ``trim-doctest-flags``.
+      A comma-separated list of doctest flags.
+
+   .. rst:directive:option:: trim-doctest-flags
+   .. rst:directive:option:: no-trim-doctest-flags
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
   .. rst:directive:option:: trim-doctest-flags
                             no-trim-doctest-flags
```

File: doc/usage/extensions/doctest.rst [hunk 4]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
...
+
+      Hide the code block in other builders.  By default it is shown as a
+      highlighted code block.
+
+   .. rst:directive:option:: trim-doctest-flags
+   .. rst:directive:option:: no-trim-doctest-flags
+
+      Whether to trim remove doctest flags (comments looking like
+      ``# doctest: FLAG, ...``) at the ends of lines and ``<BLANKLINE>`` markers
+      individually.  Default is ``trim-doctest-flags``.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst:172]: Should we have a `:default:` option to `.. rst:directive:option::`?

Comment (timhoffm) [reply doc/usage/extensions/doctest.rst:172]: Theoretically yes. In practice, I believe it's not worth it. Without checking, I assume only a minor part of options have a real default value, instead they are often flags, or the default is some form of "emtpy"/not given e.g. in the case of `:caption:`, or the default is a behavior that needs a sentence of description, and does not have a simple value to list in a `:default:` option.
So, I'm fine with writing out "Default is ..." for the cases that make sense.

File: doc/usage/extensions/doctest.rst [hunk 5]
```diff
@@ -51,6 +51,14 @@ a comma-separated list of group names.
    A setup code block.  This code is not shown in the output for other builders,
    but executed before the doctests of the group(s) it belongs to.
 
+   .. rubric:: Options
+
+   .. rst:directive:option:: skipif: condition
+      :type: text
+
+      Skip the directive if the python expression *condition* is True.
+      See :ref:`skipif`.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
      See :ref:`skipping tests conditionally <skipif>`.
```

File: doc/usage/extensions/doctest.rst [hunk 6]
```diff
@@ -59,6 +67,14 @@ a comma-separated list of group names.
 
    .. versionadded:: 1.1
 
+   .. rubric:: Options
+
+   .. rst:directive:option:: skipif: condition
+      :type: text
+
+      Skip the directive if the python expression *condition* is True.
+      See :ref:`skipif`.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
      See :ref:`skipping tests conditionally <skipif>`.
```

File: doc/usage/extensions/doctest.rst [hunk 7]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
    default set of flags is specified by the :confval:`doctest_default_flags`
    configuration variable.
 
-   This directive supports five options:
+   .. rubric:: Options
+
+   .. rst:directive:option:: hide
+
+     Hide the doctest block in other builders.  By default it is shown as a
+     highlighted doctest block.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
     Hide the doctest block in other builders.
     By default it is shown as a highlighted doctest block.
```

File: doc/usage/extensions/doctest.rst [hunk 8]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
...
+
+   .. rst:directive:option:: options: doctest flags
+      :type: comma separated list
+
+      A comma-separated list of doctest flags that apply to each example in the
+      tests.  (You still can give explicit flags per example, with doctest comments,
+      but they will show up in other builders too.)
+
+      Alternatively, you can give inline doctest options, like in doctest::
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
      Alternatively, you can give inline doctest options, like in doctest:

      .. code-block:: pycon
```

File: doc/usage/extensions/doctest.rst [hunk 9]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
...
 
    A code block for a code-output-style test.
 
-   This directive supports three options:
+   .. rubric:: Options
+
+   .. rst:directive:option:: hide
+
+      Hide the code block in other builders.  By default it is shown as a
+      highlighted code block.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
      Hide the code block in other builders.
      By default it is shown as a highlighted code block.
```

File: doc/usage/extensions/doctest.rst [hunk 10]
```diff
@@ -159,18 +200,28 @@ a comma-separated list of group names.
    The corresponding output, or the exception message, for the last
    :rst:dir:`testcode` block.
 
-   This directive supports four options:
+   .. rst:directive:option:: hide
 
-   * ``hide``, a flag option, hides the output block in other builders.  By
-     default it is shown as a literal block without highlighting.
+     Hide the doctest block in other builders.  By default it is shown as a
+     highlighted doctest block.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
     Hide the doctest block in other builders.
     By default it is shown as a highlighted doctest block.
```

File: doc/usage/extensions/doctest.rst [hunk 11]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
...
-     default it is shown as a highlighted code block.
+   .. rst:directive:option:: skipif: condition
+      :type: text
 
-   * ``trim-doctest-flags`` and ``no-trim-doctest-flags``, a flag option,
-     doctest flags (comments looking like ``# doctest: FLAG, ...``) at the
-     ends of lines and ``<BLANKLINE>`` markers are removed (or not removed)
-     individually.  Default is ``trim-doctest-flags``.
+      Skip the directive if the python expression *condition* is True.
+      See :ref:`skipif`.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
      See :ref:`skipping tests conditionally <skipif>`.
```

File: doc/usage/extensions/doctest.rst [hunk 12]
```diff
@@ -67,72 +83,97 @@ a comma-separated list of group names.
...
+      signal a blank line in the expected output.  The ``<BLANKLINE>`` is removed
+      when building presentation output (HTML, LaTeX etc.).
 
-   They will be respected when the test is run, but stripped from presentation
-   output.
+   .. rst:directive:option:: skipif: condition
+      :type: text
 
+      Skip the directive if the python expression *condition* is True.
+      See :ref:`skipif`.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
      See :ref:`skipping tests conditionally <skipif>`.
```

File: doc/usage/extensions/doctest.rst [hunk 13]
```diff
@@ -159,18 +200,28 @@ a comma-separated list of group names.
...
+
+      Whether to trim remove doctest flags (comments looking like
+      ``# doctest: FLAG, ...``) at the ends of lines and ``<BLANKLINE>`` markers
+      individually.  Default is ``trim-doctest-flags``.
+
+   .. rst:directive:option:: skipif: condition
+      :type: text
+
+      Skip the directive if the python expression *condition* is True.
+      See :ref:`skipif`.
```
Comment (AA-Turner) [inline doc/usage/extensions/doctest.rst]:
```suggestion
      See :ref:`skipping tests conditionally <skipif>`.
```

**Pull request merged**